### PR TITLE
AK/Platform.h: check for glibc

### PR DESCRIPTION
**Pull request by ddevault**
Ref #16463

### CODE REVIEW

**Review by timschumi — CHANGES_REQUESTED** [review]
`glibc` isn't the only libc that implements these functions. If we want to do something like this, dynamically checking if those functions are supported should be the way to go.

Also, again, please note the commit message format.

**Comment by timschumi** [comment]
In my opinion, if that's the main concern, that variant of the code should just be removed entirely. But as long as there is a maintained code path that uses these functions and it makes implementing it more safe for us, then why not use it everywhere where we can?

File: AK/Platform.h [hunk 1]
```diff
@@ -47,6 +47,10 @@
 #    define AK_OS_LINUX
 #endif
 
+#if defined(__GLIBC__)
+#   define AK_LIBC_GLIBC
+#endif
```
Comment (linusg) [inline AK/Platform.h:52]: Let's move this after AK_COMPILER_* instead of mixing it with the AK_OS_* macros

**Comment by ddevault** [comment]
>In my opinion, if that's the main concern, that variant of the code should just be removed entirely. But as long as there is a maintained code path that uses these functions and it makes implementing it more safe for us, then why not use it everywhere where we can?

Probably safe to remove the variant. Does it need re-entrancy? I don't think so. It's better to use portable interfaces rather than reach for reentrant interfaces on an unjustified assumption that reentrancy semantics are required. If they are required, a re-entrant implementation of the required functions should be vendored into the project.

**Comment by ddevault** [comment]
I think the new message aligns with the commit style, let me know if not so.

**Comment by trflynn89** [comment]
@ddevault We can check for the presence of these reentrant functions without #ifdef soup, which is implemented over at #16478. Can you check if that works for you / musl?

**Comment by ddevault** [comment]
I think the new approach is considerably worse. I wash my hands of the matter. I leave the choice of patch at the discretion of the maintainers.

**Comment by ADKaster** [comment]
This PR is obsolete given the changes that have been merged into master. There are still 3 remaining issues for compiling on a musl host, as I described in https://github.com/SerenityOS/serenity/pull/16323#pullrequestreview-1218671223

After those 3 are addressed, the only remaining action for https://github.com/SerenityOS/serenity/issues/16463 is to figure out where / how often / what provider to run musl CI on. It could probably slot into the nightly Azure next to the run that runs android CI.

**Pull request closed without merging**